src/encoded/types/base.py

```diff
@@ -147,11 +147,10 @@ def get_item_if_you_can(request, value, itype=None):
 def set_namekey_from_title(properties):
     name = None
     if properties.get('title'):
-        exclude = set(string.punctuation)
-        name = properties['title']
-        name = ''.join(ch for ch in name if ch not in exclude)
-        name = re.sub(r"\s+", '-', name)
-        name = name.lower()
+        exclude = set(string.punctuation.replace('-', ''))
+        name = properties['title'].replace('&', ' n ')
+        name = ''.join(ch if ch not in exclude and ch != ' ' else '-' for ch in name)
+        name = re.sub(r"[-]+", '-', name).strip('-').lower()
```

Comment:
I think this is fine. 
Since you need to use the re.sub anyway you could consider doing it all in one regex sub like so.
`exclude = string.punctuation.replace('-', '')
  name = properties['title'].replace('&', ' n ')
  regex =  = "[{}\s]+".format(exclude)
  name = re.sub(regex,  '-', name).lower()`

but I'm not sure that is any clearer and it's essentially the same amount of code so it's up to you.

Reply:
For some reason it didn't line break the code - should be 4 lines.